python_modules/dagster-graphql/dagster_graphql/schema/asset_health.py

```diff
@@ -0,0 +1,78 @@
+import graphene
+
+from dagster_graphql.schema.util import ResolveInfo
+
+
+class GrapheneAssetHealthStatus(graphene.Enum):
+    HEALTHY = "HEALTHY"
+    WARNING = "WARNING"
+    DEGRADED = "DEGRADED"
+    UNKNOWN = "UNKNOWN"
+
+    class Meta:
+        name = "AssetHealthStatus"
+
+
+class GrapheneAssetHealth(graphene.ObjectType):
+    assetHealth = graphene.NonNull(GrapheneAssetHealthStatus)
+    materializationStatus = graphene.Field(GrapheneAssetHealthStatus)
```

Comment:
Curious what people think about each sub-component of health also just having a status reported. This would power the "report card" component that shows the status for each of these indicators, but we don't have any specific information about why each indicator is in that state (like which asset check failed, etc). I think this is fine (it's not in the designs, and it seems like a user would want to click into the asset details page to get that info anyway) but curious what others think

Reply:
Hmm, I think it'd be nice to support the granularity but if theres another gql resolver that already returns that then probably don't need to create that here